### ISSUE TITLE
Can I catch the InternalMSEEDReadingWarning warning?

### ISSUE THREAD

**Issue by lileipku00** [obspy/obspy#1177]
Some of my miniseed files are problematic and raise a `InternalMSEEDReadingWarning` warning in the output.

```
/$home/anaconda/lib/python2.7/site-packages/obspy-0.10.2-py2.7-linux-x86_64.egg/obspy/mseed/core.py:356: InternalMSEEDReadingWarning: readMSEEDBuffer(): Record starting at offset 7235584 is not valid SEED. The rest of the file will not be read.
  warnings.warn(msg, InternalMSEEDReadingWarning) 
```

How do I catch it so that I can know which file is problematic?

**Comment by krischer** [comment]
That is unfortunately not possible with the current implementation as the warning is not raised in the main Python thread. We might need to change this eventually but this is actually pretty hard to pull of in a cross-platform way.

In the meanwhile you could try the CatchOutput() context manager of ObsPy and see if that can catch it.

**Comment by krischer** [comment]
This will work as soon as #1039 is merged :-) Exceptions and warning can be caught and parsed.

**Comment by QuLogic** [comment]
@krischer You should just add `Fixes #1177` in your commit messages and these will close automatically when the PR is merged.
